plugins/inputs/cosmic/cosmic.go

```diff
@@ -0,0 +1,216 @@
+package cosmic
+
+import (
+	"github.com/MissionCriticalCloud/go-cosmic/cosmic"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Cosmic struct {
+	Url       string
+	Apikey    string
+	Secretkey string
+	VerifySSL bool
+
+	Domainid string
+
+	client *cosmic.CosmicClient
+}
+
+var sampleConfig = `
+  ##
+  ## Connection parameters
+  ##
+
+  ## Cosmic API endpoint
+  url = "localhost"
+  ## The API key to use for metrics collection
+  apikey = "xxx"
+  ## The corresponding secret key
+  secretkey = "xxx"
+  ## Verify SSL certificate
+  # verifyssl = false
+
+  ##
+  ## Metric collection parameters
+  ##
+
+  ## The domain for which to collect metrics
+  # domainid = "00000000-0000-0000-0000-000000000000"
+`
+
+func (c *Cosmic) Description() string {
+	return "Gather metrics from resources available in Cosmic"
+}
+
+func (c *Cosmic) SampleConfig() string {
+	return sampleConfig
+}
+
+func (c *Cosmic) Gather(acc telegraf.Accumulator) error {
+	if c.client == nil {
+		c.client = cosmic.NewClient(c.Url, c.Apikey, c.Secretkey, c.VerifySSL)
```

Comment:
This is where I'm sticking all my go-cosmic questions :).  What about the possibility of canceling a call to this library, this can be useful during shutdown/restart/etc.  Telegraf doesn't currently have much support for `context.Context`, but this is probably something we will pass in to the `Gather` function in the future.